dbms/src/Storages/DeltaMerge/tests/gtest_dm_delta_merge_store.cpp

```diff
@@ -491,6 +492,196 @@ try
 }
 CATCH
 
+TEST_P(DeltaMergeStoreRWTest, WriteCrashBeforeWalWithoutCache)
+try
+{
+    const ColumnDefine col_str_define(2, "col2", std::make_shared<DataTypeString>());
+    const ColumnDefine col_i8_define(3, "i8", std::make_shared<DataTypeInt8>());
+    {
+        auto table_column_defines = DMTestEnv::getDefaultColumns();
+        table_column_defines->emplace_back(col_str_define);
+        table_column_defines->emplace_back(col_i8_define);
+
+        store = reload(table_column_defines);
+    }
+
+    {
+        // check column structure
+        const auto & cols = store->getTableColumns();
+        ASSERT_EQ(cols.size(), 5UL);
+        const auto & str_col = cols[3];
+        ASSERT_EQ(str_col.name, col_str_define.name);
+        ASSERT_EQ(str_col.id, col_str_define.id);
+        ASSERT_TRUE(str_col.type->equals(*col_str_define.type));
+        const auto & i8_col = cols[4];
+        ASSERT_EQ(i8_col.name, col_i8_define.name);
+        ASSERT_EQ(i8_col.id, col_i8_define.id);
+        ASSERT_TRUE(i8_col.type->equals(*col_i8_define.type));
+    }
+
+    const size_t num_rows_write = 128;
+    {
+        // write to store
+        Block block;
+        {
+            block = DMTestEnv::prepareSimpleWriteBlock(0, num_rows_write, false);
+            // Add a column of col2:String for test
+            block.insert(DB::tests::createColumn<String>(
+                createNumberStrings(0, num_rows_write),
+                col_str_define.name,
+                col_str_define.id));
+            // Add a column of i8:Int8 for test
+            block.insert(DB::tests::createColumn<Int8>(
+                createSignedNumbers(0, num_rows_write),
+                col_i8_define.name,
+                col_i8_define.id));
+        }
+        db_context->getSettingsRef().dt_segment_delta_cache_limit_rows = 8;
+        FailPointHelper::enableFailPoint(FailPoints::exception_after_write_blob);
+        try
+        {
+            store->write(*db_context, db_context->getSettingsRef(), block);
+        }
+        catch (DB::Exception & e)
+        {
+            if (e.code() != ErrorCodes::FAIL_POINT_ERROR)
```

Comment:
Add a check to make sure the `catch` clause is executed to avoid the fail point not working.